libs/auth-api-lib/src/lib/personal-representative/services/personalRepresentative.service.ts

```diff
@@ -237,17 +272,53 @@ export class PersonalRepresentativeService {
           },
         )
 
-        await this.personalRepresentativeRightModel.bulkCreate(rightCodes, {
-          transaction: t,
-        })
+        const createdPRR =
+          await this.personalRepresentativeRightModel.bulkCreate(rightCodes, {
+            transaction: t,
+          })
+
+        // Loop through the created PersonalRepresentativeRight and create a PersonalRepresentativeDelegationType for each,
+        // this is done to use the same id for the PersonalRepresentativeDelegationType as the PersonalRepresentativeRight
+        await Promise.all(
+          createdPRR.map(async (c) => {
+            try {
+              const created =
+                await this.personalRepresentativeDelegationTypeModel.create(
+                  {
+                    id: c.id,
+                    personalRepresentativeId: newPr.id,
+                    delegationTypeId: `PersonalRepresentative:${c.rightTypeCode}`,
+                  },
+                  { transaction: t },
+                )
+
+              return created.id
+            } catch (error) {
+              throw new BadRequestException(
+                `Error creating personal representative delegation type: ${error}`,
+              )
+            }
```

Comment:
This can be a misleading error classification and devops prefer structured error log object, meaning not inline string it in the message just add it as as a object param.

If the `rightTypeCode` does not exists in the `delegation_type` table, or the PR does not exists then this is correctly a users bad request. But if we have some random db failure (connection exhaustion, timeouts etc.) this should be Internal Server error, causing 500.

This catches everything and re-throws it as a 400 bad request. 

You could try checking if the error contains some foreign key reference error and then throw bad request other wise throw the same error.

```suggestion
            } catch (error) {
              if (error somehow contains foreign key ref error) {
                throw new BadRequestException(
                  'Error creating personal representative delegation type',
                  error
                )
              }
              else {
                 throw error
              }
            }
```